### ISSUE TITLE
NoClassDefFoundError

### ISSUE THREAD

**Issue by shiredude95** [kuri65536/python-for-android#153]
Im facing a Class not found error when I try to incorporate the files into my project and compile them.
Im running the python bin as a background service, but the app crashes when setting up Android Proxy as it shows ClassNotFound for FacadeConfiguration.
Everything works fine if I run it as a standalone app, but when I try to integrate it I get this error. Note all the reequired libraries are in the path. (the so files and the jars)

```
     Caused by: java.lang.NoClassDefFoundError: Failed resolution of: Lcom/googlecode/android_scripting/facade/FacadeConfiguration;
        at com.googlecode.android_scripting.AndroidProxy.<init>(AndroidProxy.java:53)
        at com.dummy.android.python32.BackgroundScriptService.startMyMain(BackgroundScriptService.java:209)
        at com.dummy.android.python32.BackgroundScriptService.access$000(BackgroundScriptService.java:70)
        at com.dummy.android.python32.BackgroundScriptService$startMyAsyncTask.doInBackground(BackgroundScriptService.java:166)
        at com.dummy.android.python32.BackgroundScriptService$startMyAsyncTask.doInBackground(BackgroundScriptService.java:159)
        at android.os.AsyncTask$2.call(AsyncTask.java:333)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636) 
        at java.lang.Thread.run(Thread.java:764) 
     Caused by: java.lang.ClassNotFoundException: Didn't find class "com.googlecode.android_scripting.facade.FacadeConfiguration" on path: DexPathList[[zip file "/data/app/com.dummy-b7cnGxmpsPXX4DCk5rJbOA==/base.apk"],nativeLibraryDirectories=[/data/app/com.dummy-b7cnGxmpsPXX4DCk5rJbOA==/lib/arm, /data/app/com.dummy-b7cnGxmpsPXX4DCk5rJbOA==/base.apk!/lib/armeabi-v7a, /system/lib, /vendor/lib]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:125)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at com.googlecode.android_scripting.AndroidProxy.<init>(AndroidProxy.java:53) 
        at com.dummy.android.python32.BackgroundScriptService.startMyMain(BackgroundScriptService.java:209) 
        at com.dummy.android.python32.BackgroundScriptService.access$000(BackgroundScriptService.java:70) 
        at com.dummy.android.python32.BackgroundScriptService$startMyAsyncTask.doInBackground(BackgroundScriptService.java:166) 
        at com.dummy.android.python32.BackgroundScriptService$startMyAsyncTask.doInBackground(BackgroundScriptService.java:159) 

```

**Comment by kuri65536** [comment]
I could not realize your situation...

- I don't have experience to create the app which has builtin SL4A program.
- Lack of information

If you want to just run python in your application,
Android proxy is not needed and you can create the server in your app.

**Comment by shiredude95** [comment]
I did manage to resolve the issue. While trying to integrate the package into an existing package there are 2 ways: one an either make a library and import it or directly make a package in the main project and use classes within. The first case would not work here as we have some resources in the assets folder which cannot be imported from a library project. So we are left with creating a package and using the classes within. This is what I tried to do, but unfortunately android kept throwing the class not found error. I decompiled the script.jar and found that it does not indeed have a file called FacadeConfiguration.  The jvm looks for this specific path while executing the code "com.googlecode.android_scripting.facade.FacadeConfiguration". When I modified my project structure in such a way that the facadeconfig file provided in the package had the same package name as one requested by the android proxy file, jvm was able to find it and execute the code. This was after numerous trial and error attempts though. So a small note wrt this would be very helpful for new users.
